### PR TITLE
#40: `FromDataframeSampleSizer` should not create empty samples

### LINKED ISSUE #40: `FromDataframeSampleSizer` should not create empty samples
## Describe the bug
Sometimes `FromDataframeSampleSizer.performRowSizing` returns 0 even on non-empty `DataFrame` because the sample create is empty.
The sample should not be empty on a non-empty `DataFrame`.

## To Reproduce
Run the unit tests in `FromDataframeSampleSizerTest`, occasionally they fail.

### PR DESCRIPTION
**Pull request by benedeki**
* ensured that sample from non-empty DataFrame is never empty
* changed `performRowSizing` to accept optional record count as Long instead of Int

Closes #40

### CODE REVIEW

File: src/main/scala/za/co/absa/spark/partition/sizing/sizer/FromDataframeSampleSizer.scala [hunk 1]
```diff
@@ -25,28 +25,33 @@ import za.co.absa.spark.partition.sizing.utils.RowSizer
...
-        .limit(sampleSize)
-        .collect()
-        .map(RowSizer.rowSize)
-
-      if(sampleSizes.isEmpty) {
-        0L
-      } else {
-        ceilDiv(sampleSizes.sum, sampleSizes.length)
-      }
+      val sampleFraction =computeSampleFraction(df, dfRecordCount)
```
Comment (jakipatryk) [inline src/main/scala/za/co/absa/spark/partition/sizing/sizer/FromDataframeSampleSizer.scala:51]: Missing space after `=`.

**Review by jakipatryk — DISMISSED** [review]
LGTM (just checked the code).

File: src/main/scala/za/co/absa/spark/partition/sizing/sizer/FromDataframeSampleSizer.scala (hunk 1 above)
Comment (benedeki) [reply src/main/scala/za/co/absa/spark/partition/sizing/sizer/FromDataframeSampleSizer.scala:51]: 👍 
```suggestion
      val sampleFraction = computeSampleFraction(df, dfRecordCount)
```

**Review by jakipatryk — DISMISSED** [review]

**Review by dk1844 — APPROVED** [review]
LGTM
- [x] code reviewed
- [x] pulled
- [x] built
- [x] run a series of `sbt test`s to check flakiness.

I have also devised a test that was reasonably sure to fail on `master` while now passes on this branch, I offer it to you so can consider including it in `FromDataframeSampleSizerTest`:

```scala
  test("test non-empty input has a positive size (non-empty) sample is generated from non-empty input)") {
    val sizer1 = new FromDataframeSampleSizer(1)
    // need to do a few retries because sampling is randomly imprecise in count:
    // sampling 1 out of 2 can lead to 0-sized sample!
    (0 to 10).foreach { _ =>
      assert(sizer1.performRowSizing(simpleDf) > 0)
    }
  }
```

**Comment by benedeki** [comment]
> 
> I have also devised a test that was reasonably sure to fail on `master` while now passes on this branch, I offer it to you so can consider including it in `FromDataframeSampleSizerTest`:
> 
> ```scala
>   test("test non-empty input has a positive size (non-empty) sample is generated from non-empty input)") {
>     val sizer1 = new FromDataframeSampleSizer(1)
>     // need to do a few retries because sampling is randomly imprecise in count:
>     // naive usage of df.sample for 1 out of 2 can lead to 0-sized sample!
>     (1 to 5).foreach { _ =>
>       assert(sizer1.performRowSizing(simpleDf) > 0)
>     }
>   }
> ```

Added the suggested test, smart!

**Review by miroslavpojer — APPROVED** [review]
- [x] code reviewed
- [x] pulled
- [x] built
- [x] run

**Pull request merged**